ml-image/scripts/faces_clustering.py

```diff
@@ -0,0 +1,104 @@
+from PIL import Image
+import numpy as np
+from facenet_pytorch import InceptionResnetV1
+from dlib import chinese_whispers_clustering, vector
+import statistics
+
+import torch
+from torch.utils.data import DataLoader
+from torchvision import transforms
+
+from .utils import ImageDataset
+    
+class Embedding_model():
+    def __init__(self, model = "facenet", device = 'cpu'):
+        self.device = device
+        self.model_name = model
+
+        match self.model_name:
+            case "facenet":
+                self.model = InceptionResnetV1(pretrained='vggface2').eval().to(self.device)
+            case _:
+                raise ValueError("Model not supported")
+
+    def get_embedding(self, batch_size: int, detected_faces: list):
+        # Transformation for model compliance
+        match self.model_name:
+            case "facenet":
+                face_transform = transforms.Compose([
+                transforms.ToTensor(),          # Convertir en tenseur
+                transforms.Resize((160, 160)),  # Taille attendue par FaceNet
+                transforms.Normalize([0.5], [0.5])  # Normalisation [-1, 1]
+                ])
+            case _:
+                raise ValueError("Model not supported")
+
+        dataset = ImageDataset(detected_faces, face_transform)
+        dataloader = DataLoader(dataset, batch_size=batch_size, num_workers=0)
+
+        embeddings = []
+
+        with torch.no_grad(): 
+            for batch in dataloader:
+                batch = batch.to(self.device)
+                emb = self.model(batch)
+                embeddings.append(emb.cpu())
+
+        if len(embeddings) == 0:
+            return np.array([])
+
+        embeddings = torch.cat(embeddings)  # Concatenate all embeddings
+        embeddings_array = embeddings.numpy() # Convert to numpy array for use in clustering algorithms
+
+        return embeddings_array
+
+class Faces_clustering():
+    def __init__(self, model = "chinese_whispers", **kwargs):
```

Comment:
peut etre mettre directement threshold plutot que kwargs ?
```suggestion
    def __init__(self, model = "chinese_whispers", threshold: float,  **kwargs):
            self.model = model
            self.threshold = threshold
```